osquery/tables/system/windows/logical_drives.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+#include <osquery/tables.h>
+
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genLogicalDrives(QueryContext& context) {
+  QueryData results;
+
+  WmiRequest wmiLogicalDiskReq(
+      "select DeviceID, DriveType, FreeSpace, Size, FileSystem from "
+      "Win32_LogicalDisk");
+  std::vector<WmiResultItem>& wmiResults = wmiLogicalDiskReq.results();
+  for (unsigned int i = 0; i < wmiResults.size(); ++i) {
+    Row r;
+    unsigned int driveType = 0;
+    std::string deviceId, freeSpaceStr, sizeStr;
+    wmiResults[i].GetString("DeviceID", deviceId);
+    wmiResults[i].GetUnsignedInt32("DriveType", driveType);
+    wmiResults[i].GetString("FreeSpace", freeSpaceStr);
+    wmiResults[i].GetString("Size", sizeStr);
+    wmiResults[i].GetString("FileSystem", r["file_system"]);
+
+    r["device_id"] = deviceId;
+    r["free_space"] = BIGINT(freeSpaceStr);
+    r["size"] = BIGINT(sizeStr);
+
+    switch (driveType) {
+    default:
+      r["type"] = TEXT("Unknown");
+      break;
+    case 1:
+      r["type"] = TEXT("No Root Directory");
+      break;
+    case 2:
+      r["type"] = TEXT("Removable Disk");
+      break;
+    case 3:
+      r["type"] = TEXT("Local Disk");
+      break;
+    case 4:
+      r["type"] = TEXT("Network Drive");
+      break;
+    case 5:
+      r["type"] = TEXT("Compact Disc");
+      break;
+    case 6:
+      r["type"] = TEXT("RAM Disk");
+      break;
+    }
+
+    r["boot_partition"] = INTEGER(0);
```

Comment:
Is there any way to make this real?

Reply:
Can you expand on this? I don't think I understand what you're asking for.

Reply:
Sorry, I made a mistake before and thought this was the only use of `boot_partition`. :+1: